Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -449,306 +361,73 @@ void
-                        }
-                    }
-                }
-            }
-        }, amrex::Scan::Type::exclusive);
 
+    auto const &cell_size = CellSize(maxLevel());
 
-    }
-    // Do the extensions in the y-plane
-    idim = 1;
+    const amrex::Real dx = cell_size[0];
+    const amrex::Real dy = cell_size[1];
+    const amrex::Real dz = cell_size[2];
 
-    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+    // Do the extensions
+    for(int idim = 0; idim < AMREX_SPACEDIM; idim++){
+        for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
 
-        amrex::Box const &box = mfi.validbox();
+            amrex::Box const &box = mfi.validbox();
 
-        auto const &Sy = m_face_areas[maxLevel()][idim]->array(mfi);
-        auto const &flag_ext_face_y = m_flag_ext_face[maxLevel()][idim]->array(mfi);
-        auto const &flag_info_face_y = m_flag_info_face[maxLevel()][idim]->array(mfi);
-        auto &borrowing_y = (*m_borrowing[maxLevel()][idim])[mfi];
-        auto const &borrowing_y_inds_pointer = borrowing_y.inds_pointer.array();
-        auto const &borrowing_y_size = borrowing_y.size.array();
-        amrex::Long ncells = box.numPts();
-        int* borrowing_y_inds = borrowing_y.inds.data();
-        FaceInfoBox::Neighbours* borrowing_y_neigh_faces = borrowing_y.neigh_faces.data();
-        amrex::Real* borrowing_y_area = borrowing_y.area.data();
-        int& vecs_size_y = borrowing_y.vecs_size;
-
-        auto const &Sy_mod = m_area_mod[maxLevel()][idim]->array(mfi);
-        const auto &lx = m_edge_lengths[maxLevel()][0]->array(mfi);
-        const auto &lz = m_edge_lengths[maxLevel()][2]->array(mfi);
-        const amrex::Real dx = cell_size[0];
-        const amrex::Real dz = cell_size[2];
-
-        vecs_size_y = amrex::Scan::PrefixSum<int>(ncells,
-            [=] AMREX_GPU_DEVICE (int icell) {
+            auto const &S = m_face_areas[maxLevel()][idim]->array(mfi);
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+            auto const &flag_info_face = m_flag_info_face[maxLevel()][idim]->array(mfi);
+            auto &borrowing = (*m_borrowing[maxLevel()][idim])[mfi];
+            auto const &borrowing_inds_pointer = borrowing.inds_pointer.array();
+            auto const &borrowing_size = borrowing.size.array();
+            amrex::Long ncells = box.numPts();
+            int* borrowing_inds = borrowing.inds.data();
+            FaceInfoBox::Neighbours* borrowing_neigh_faces = borrowing.neigh_faces.data();
+            amrex::Real* borrowing_area = borrowing.area.data();
+            int& vecs_size = borrowing.vecs_size;
+
+            auto const &S_mod = m_area_mod[maxLevel()][idim]->array(mfi);
+
+            const auto &lx = m_edge_lengths[maxLevel()][0]->array(mfi);
+            const auto &ly = m_edge_lengths[maxLevel()][1]->array(mfi);
+            const auto &lz = m_edge_lengths[maxLevel()][2]->array(mfi);
```

Comment:
I don't think anything is needed given the way I initialized these multifabs (m_edge_lengths[maxLevel()][1] is all zeros and the other two contain the x and z edge lengths).
More care is needed when we access all these multifabs because the z-axis is in second position, but that's what I'm taking care of in GetNeigh, SetNeigh and ComputeSStab